### PR TITLE
genfstab: add page

### PR DESCRIPTION
**Pull request by peterbabic**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

File: pages/linux/genfstab.md
Comment (navarroaxel) [inline pages/linux/genfstab.md:4]:
```suggestion
> More information: <https://man.archlinux.org/man/extra/arch-install-scripts/genfstab.8>.
```

File: pages/linux/genfstab.md
Comment (navarroaxel) [inline pages/linux/genfstab.md:10]: Maybe this is better:
```suggestion
- Display `fstab` output using partitions' UUID:
```

File: pages/linux/genfstab.md
Comment (navarroaxel) [inline pages/linux/genfstab.md:12]:
```suggestion
`genfstab -U {{path/to/mount_point}}`
```

File: pages/linux/genfstab.md
Comment (navarroaxel) [inline pages/linux/genfstab.md:8]:
```suggestion
`genfstab -L {{path/to/mount_point}}`
```

File: pages/linux/genfstab.md
Comment (navarroaxel) [inline pages/linux/genfstab.md:16]:
```suggestion
`genfstab -U {{/mnt}} >> {{/mnt/etc/fstab}}`
```

File: pages/linux/genfstab.md
Comment (navarroaxel) [inline pages/linux/genfstab.md:20]:
```suggestion
`genfstab -U {{path/to/mount_point}} | sudo tee -a /etc/fstab`
```

**Review by peterbabic — COMMENTED** [review]
add maintainer suggestions

**EDIT:** do not know what to do, still reluctantly learning Github 
**EDIT2:** somehow aded maintainer suggestions into "batch" and commited them, hopefully this was expected

File: pages/linux/genfstab.md
Comment (bl-ue) [inline pages/linux/genfstab.md:6]: This is a little confusing&mdash;where does LABEL come from?

File: pages/linux/genfstab.md
Comment (bl-ue) [inline pages/linux/genfstab.md:14]: Instead of giving superuser notice, you can just add `sudo` to the command (see additional suggestion below):
```suggestion
- Generate an fstab file:
```

File: pages/linux/genfstab.md
Comment (bl-ue) [inline pages/linux/genfstab.md:16]: See my suggestion above.
```suggestion
`sudo genfstab -U {{/mnt}} >> {{/mnt/etc/fstab}}`
```

**Review by bl-ue — CHANGES_REQUESTED** [review]
Hi @peterbabic, thank you for the new page! I've left some comments below for you to review.

File: pages/linux/genfstab.md
Comment (peterbabic) [reply pages/linux/genfstab.md:16]: This suggestion will fail, for sudo there is the line just below.
Thiis line in specifically used when superuser and it is the 99% of it's use cases, I would definitely not edit it in this way.

File: pages/linux/genfstab.md
Comment (peterbabic) [reply pages/linux/genfstab.md:14]: An fstab? I do not understand the edit.

File: pages/linux/genfstab.md
Comment (peterbabic) [reply pages/linux/genfstab.md:6]: The LABEL is a filesystem volume identifier, usually assigned (but not limited to) during the volume creation. More information: https://wiki.archlinux.org/index.php/Persistent_block_device_naming#by-label

File: pages/linux/genfstab.md
Comment (bl-ue) [reply pages/linux/genfstab.md:14]: "an fstab file".  I assume "fstab" is generally pronounced "ef-es-tab"&mdash;@navarroaxel am I correct about this?

File: pages/linux/genfstab.md
Comment (bl-ue) [reply pages/linux/genfstab.md:16]: By "as superuser", do you mean after having run something like `su`?

Comment (bl-ue) [reply pages/linux/genfstab.md:16]: If it's required to run the command as superuser as you say, why not prefix it with `sudo`?  We do that in many other pages, particularly in low-level file-system utilities, such as [`partx.md`](https://github.com/tldr-pages/tldr/blob/master/pages/linux/partx.md).

**Comment by peterbabic** [comment]
Unfortunately I do not agree wit any of the changes @bl-ue suggested, not meaning to be rude by any way. I have provided explanations to all of them.

File: pages/linux/genfstab.md
Comment (bl-ue) [reply pages/linux/genfstab.md:6]: Then can you update either the description or the command, because it is not clear what "LABEL" has to do with the example command itself.

Maybe something like this?
```suggestion
- Generate the output based on a label:
```
and then change the example from `path/to/mount_point` to `path/to/label`?

I don't use Arch Linux, so feel free to correct me!

File: pages/linux/genfstab.md
Comment (peterbabic) [reply pages/linux/genfstab.md:16]: To your First question, yes, for instance `su` is necessary to elevate privileges.

To your second question, the command will fail because of the redirection operator `>>`. Prefixing the whole line with with `sudo` will not work. To use redirection operator with elevated privileges, a different approach is necessary.

https://linuxize.com/post/bash-append-to-file/#append-to-a-file-using-the-redirection-operator-
https://www.cyberciti.biz/faq/sudo-append-data-text-to-file-on-linux-unix-macos/

**Comment by bl-ue** [comment]
@peterbabic no problem! Sometimes we have difficulty communicating through a purely textual medium :)  I've responded to your comments.

File: pages/linux/genfstab.md
Comment (bl-ue) [reply pages/linux/genfstab.md:16]: Ah, I see!  Makes perfect sense now.  Please disregard this suggestion.

File: pages/linux/genfstab.md
Comment (bl-ue) [reply pages/linux/genfstab.md:14]: It's better English to use "an" before a word like "fstab", the way it's generally pronounced.  Now of course, people have different pronunciations but I believe "an" would be a better fit here.
```suggestion
- Generate an fstab file, requires being a superuser:
```
Please also disregard the first suggestion, as I understand the superuser situation now :)

File: pages/linux/genfstab.md
Comment (bl-ue) [inline pages/linux/genfstab.md:16]: How about this:
```suggestion
`genfstab -U {{path/to/mount_point}} >> {{/mnt/etc/fstab}}`
```

**Comment by peterbabic** [comment]
@bl-ue no poblem as well! Thanks for suggestions. Did not know about the nature of "an ef-es-tab file" 😄

File: pages/linux/genfstab.md
Comment (peterbabic) [reply pages/linux/genfstab.md:16]: I would rather change the description to: 

- Most usual way to generate an fstab file, requires being a superuser:

And left the code the way it is. What do you think?

Comment (bl-ue) [reply pages/linux/genfstab.md:16]: 1. IMO, changing the description isn't necessary. I think it's good already 👍🏻 
2. `/mnt` is inconsistent it seems with the rest, that may be intentional (I'm no Arch user :smile:)

Comment (bl-ue) [reply pages/linux/genfstab.md:16]: Actually, I think it would be better to make this example more consistent with the last one, which would solve the superuser problem, by using `sudo tee`:
```suggestion
`genfstab -U {{path/to/mount_point}} | sudo tee -a {{/mnt/etc/fstab}}`
```

Comment (bl-ue) [reply pages/linux/genfstab.md:16]: If you're sure that `/mnt` is better, even though it's inconsistent, let it be that way. I trust you know better :)

Comment (navarroaxel) [reply pages/linux/genfstab.md:16]: @bl-ue in this example @peterbabic tried to explain the most common use case for this command as a step in the Arch Linux installation https://wiki.archlinux.org/index.php/installation_guide#Fstab. Nobody uses `sudo tee` 😄 because the installer is a live ISO and you run it with `root` user, command by command.

Comment (peterbabic) [reply pages/linux/genfstab.md:16]: Look, we can even discard the first and the second example. The third and the fourth are the only ones that I have personally ever needed and will serve for most people.

Comment (navarroaxel) [reply pages/linux/genfstab.md:16]: Yeah, but the first 2 examples are useful to explain how the command works and some arguments.

Comment (peterbabic) [reply pages/linux/genfstab.md:16]: Yes exactly, that is why I have added there. I was following the contribution guidelines.

Comment (bl-ue) [reply pages/linux/genfstab.md:16]: Okay guys, great!  @navarroaxel cool that you're an Arch Linux guy.  That's good to know.

> Nobody uses `sudo tee` 😄 because the installer is a live ISO and you run it with root user, command by command.

😄😄😄😄😄

File: pages/linux/genfstab.md
Comment (bl-ue) [inline pages/linux/genfstab.md:14]: Perhaps this will read better?
```suggestion
- Generate an fstab file, requires root permissions:
```

File: pages/linux/genfstab.md
Comment (bl-ue) [inline pages/linux/genfstab.md:6]: Perhaps this will read better?
```suggestion
- Generate output based on a label:
```

**Review by bl-ue — CHANGES_REQUESTED** [review]
Okay, thanks for the help @navarroaxel.  @peterbabic I have only two more comments.

**Comment by peterbabic** [comment]
I have added another commit where I believe the changes are the most consistent. I have also removed backticks around `fstab` as it is word, not a code nor a path.

**Comment by peterbabic** [comment]
Maybe the last thing I would check is if *volume* is the right word, as *filesystem*, *block device*, *disk*, *partition* or *disk partition* would all suit as well depending on use case.

 I would keep *volume* there.

image

**Comment by bl-ue** [comment]
@peterbabic - Looks good to me, except the sentence "A usual way to generate an fstab file, requires root permissions:" is inconsistent with the rest of our pages.  If that is the most common example, then I think you should:
1. Move the example to the top, so it's what the user sees first, and
2. Change the example description to "Generate an fstab file (requires root permissions):"

File: pages/linux/genfstab.md
Comment (sbrl) [reply pages/linux/genfstab.md]: Yeah, that's correct grammar there @bl-ue.

File: pages/linux/genfstab.md
Comment (sbrl) [reply pages/linux/genfstab.md:16]: Hrm, I see the problem here. I would be inclined to go with @peterbabic's suggestion here, with something like this:

```suggestion
`genfstab -U {{path/to/mount_point}} | sudo tee -a /etc/fstab`
```

The reason for this is that by convention in other pages, we use `sudo` to indicate that a given command needs to be run as root - so I'd prefer to use that same convention here too if possible.

Also, welcome to GitHub and tldr-pages, @peterbabic!

**Review by sbrl — APPROVED** [review]
Looks pretty much good to go to me - I think we can probably merge this.

I've left a comment on the `requires root permissions` issue, but that doesn't necessarily need to block the merge of this PR - and I can see some significant discussion has already happened - so I'm leaning on the side of getting this merged sooner rather than later.

**Review by navarroaxel — APPROVED** [review]

File: pages/linux/genfstab.md
Comment (bl-ue) [reply pages/linux/genfstab.md:16]: @sbrl this has already been resolved on another long thread on this same line :) - https://github.com/tldr-pages/tldr/pull/5215#discussion_r567993244

**Comment by bl-ue** [comment]
@sbrl that was actually already resolved :) - https://github.com/tldr-pages/tldr/pull/5215#discussion_r569025992

**Review by bl-ue — APPROVED** [review]
Thank you @peterbabic, and welcome to tldr-pages!

**Comment by sbrl** [comment]
Ok, whatever was done there is fine then :-)

**Comment by bl-ue** [comment]
We picked not to use `sudo` because that command is normally run in a boot environment where `tee` may not be available.

**Pull request merged**
